### PR TITLE
feat: linter timers

### PR DESCRIPTION
**Pull request by jaredallard**
## What this PR does / why we need it

This PR improves the linters to show how long they took, as well as changed their activiation criteria to be based on file extensions/other criteria.

Example:

gif of it running

## Jira ID

[XX-XX]

### CODE REVIEW

File: root/Makefile [hunk 1]
```diff
@@ -96,9 +96,14 @@ pre-commit: fmt
 .PHONY: build
 build:: pre-build gobuild
 
+## lint:            run code linters
+.PHONY: lint
+lint:: pre-test
```
Comment (george-e-shaw-iv) [inline root/Makefile:101]: should you just make a new hook, `pre-lint`?

Comment (jaredallard) [reply root/Makefile:101]: Prolly, for compat I'll run pre-test tho

**Review by george-e-shaw-iv — APPROVED** [review]
this is cool - two general comments that you can either decide to do or not do, won't hurt my feelings:

1. i meant to add what file types each of them run on in a similar fashion to what i did with `make fmt`, but i just forgot to do so. if you wanna add that i think it'd also be nice for subtle education
2. can we change the name of the linter script, `validate.sh` to `lint.sh`?

File: root/Makefile [hunk 2]
```diff
@@ -96,9 +97,15 @@ pre-commit: fmt
 .PHONY: build
 build:: pre-build gobuild
 
+## lint:            run code linters
+.PHONY: lint
+# Note: We run pre-test for compat. Remove on the next breaking change.
```
Comment (george-e-shaw-iv) [inline root/Makefile:102]: definitely make a ticket for this and add it to the breaking changes dashboard julia has. this one has a high chance of being forgotten about

**Pull request merged**